pkg/api/vm/handler.go

```
@@ -717,6 +719,43 @@ func (h *vmActionHandler) createVMImages(templateVersion *harvesterv1.VirtualMac
 	return nil
 }
 
+// Since the LH volume creation and replica scheduling are asynchronous
+// Even the CreateVolume csi call success, the replica scheduling may still fail
+// However for the sc with VolumeBindingMode is Immediate
+// we could check annotation for corresponding error if it exists
+func (h *vmActionHandler) checkForConventionSC(pvc *corev1.PersistentVolumeClaim) error {
+	sc, err := h.storageClassCache.Get(*pvc.Spec.StorageClassName)
+	if err != nil {
+		return err
+	}
+
+	// Even harvester specify default StorageClass,
+	// we check the sc existence for robustness
+	if sc == nil {
```

When the sc is nil, most CSI driver will use the default SC.
I checked with your comment above if we assume the sc should not be empty.
We may need to return the error here?

---

If the sc is not specified during pvc creation, we would find the sc points to default sc, rather than nil. I added this check for robustness from https://github.com/harvester/harvester/pull/3869#pullrequestreview-1424150708.

Your comment is a fair point. I would change the checking here. Thanks

---

I do some tests here. Looks like we could not have nil StorageClassName, so we should check with an empty string and return an error because we want the sc to exist. Should we?

---

Agree, I would update the checking per your suggestion.

---

Modify checking: return error if sc is found to be nil.